File: apps/spotter/macos/spotter-macOS/Shell/Shell.m
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(Shell, RCTEventEmitter)

RCT_EXTERN_METHOD(execute:(NSString) command
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

@end

File: macos/spotter-macOS/Shell/Shell.swift
import Foundation

@objc(Shell)
class Shell: RCTEventEmitter {

  @objc
  func execute(_ command: String,
                      resolver resolve: RCTPromiseResolveBlock,
                      rejecter reject: RCTPromiseRejectBlock) -> Void {
    do {
      let output = try shellOut(to: command, at: "/")
      resolve(output)
    } catch {
      let error = error as! ShellOutError
      print(error.message)
      print(error.output)
      reject("code", error.message, error)
    }
  }

  override func supportedEvents() -> [String]? {
    return []
  }

  @objc override static func requiresMainQueueSetup() -> Bool {
    return false
  }

}
